src/plugins/notebook/components/Notebook.vue

```diff
@@ -377,25 +406,74 @@ export default {
+
             sectionKeys.forEach(sectionKey => {
                 const pages = entries[sectionKey];
                 const pageKeys = Object.keys(pages);
+                const section = this.getSection(sectionKey);
+                let resultMetadata = {
+                    originalSearchText,
+                    sectionHit: section.name && section.name.toLowerCase().includes(searchTextLower)
+                };
+
                 pageKeys.forEach(pageKey => {
                     const pageEntries = entries[sectionKey][pageKey];
+                    const page = this.getPage(section, pageKey);
+                    resultMetadata.pageHit = page.name && page.name.toLowerCase().includes(searchTextLower);
+
                     pageEntries.forEach(entry => {
-                        if (entry.text && entry.text.toLowerCase().includes(this.search.toLowerCase())) {
-                            const section = this.getSection(sectionKey);
-                            output.push({
+                        const entryHit = entry.text && entry.text.toLowerCase().includes(searchTextLower);
+
+                        // any entry hit goes in, it's the most unique of the hits
+                        if (entryHit) {
+                            resultMetadata.entryHit = entryHit;
+
+                            output.push(objectCopy({
+                                metadata: resultMetadata,
                                 section,
-                                page: this.getPage(section, pageKey),
+                                page,
                                 entry
-                            });
+                            }));
                         }
+
+
                     });
+                    // all entries checked, now in pages,
+                    // if page hit, but not in results, need to add
+                    if (resultMetadata.pageHit && this.notInResults('page', page, output)) {
+                        resultMetadata.entryHit = false;
+
+                        output.push(objectCopy({
+                            metadata: resultMetadata,
+                            section,
+                            page
+                        }));
+                    }
+
                 });
+                // all pages checked, now in sections,
+                // if section hit, but not in results, need to add and default page
+                if (resultMetadata.sectionHit && this.notInResults('section', section, output)) {
+                    resultMetadata.entryHit = false;
+                    resultMetadata.pageHit = false;
+
+                    output.push(objectCopy({
+                        metadata: resultMetadata,
+                        section,
+                        page: this.getPage(section, pageKeys[0])
+                    }));
+                }
+
             });
 
-            return output;
+            this.searchResults = output;
+        },
+        notInResults(type, item, results) {
```

Comment:
hmm, I think we can track `anyPageHit` and `anyEntryHit` and use it to in place of `notInResults`, so it saves result looping?

something like,

```
let anyPageHit = false;
...
if (pageHit) {
    anyPageHit = true;
}
.....
let anyEntryHit = false;
.....
if (entryHit) {
      anyEntryHit = entryHit;
}

if (resultMetadata.pageHit && !anyEntryHit) {

if (resultMetadata.sectionHit && !anyPageHit) {

```

Reply:
Ah, I gotcha. Let me take a look at that!

Reply:
Ok, updated with this type of logic... double check for me, I smoke-tested and it looked good, but I had to add an extra bool for tracking entry hits for sections and entry hits for pages (have to check those differently I believe) also for section hit checks, need to check if there is any page OR entry, as an entry hit would put the section in the results already. was able to get rid of that function though.